src/basic/cgroup-util.c

```diff
@@ -1948,6 +1948,9 @@ int cg_slice_to_path(const char *unit, char **ret) {
         assert(unit);
         assert(ret);
 
+        if (!slice_name_is_valid(unit))
```

Comment:
hmm, i am not entirely sure how i feel about this patch... so far the cgroup tree (modulo delegations) and the slice tree were bijective, you could translate a slice name to the cgroup path and back without any kind of additional information, and in a strictly unique way. And that's a great property, because of its simplicity and the fact that you never need to consult PID 1 about additional information regarding this mapping...

Then there's the other issue: automount and mount units have very similar semantics in this regard: for them too the path is in the fs and the unit name are bijective. And for them too people asked for templating functionality. And if we add it for slices we should add it for mount and automount units too i figure...

Now, I agree that templating makes sense, but we should really try to salvage some of the bijectiveness...

One option that i have been thinking about is to say for *all* unit types the unit with the name foo@bar.quux has an implicit alias of foo-bar.quux, and vice versa. i.e. that would also mean that sshd@foobar.service would gain an implicitl alias sshd-foobar.service. Moreover systemd-resolved.service would gain an implicit alias of systemd@resolved.service. Of course, not all of these aliases make much sense, but it would unify things relatively nicely I think.

With such a rule we would declare that every name that contains one of `@` or `-` always exists in two equivalent forms: the "template form" and the "normal form". 

But this would still leave some questions open: how do we determine which form to show by default for a unit?